src/misc/guillotine.ts

```diff
@@ -0,0 +1,235 @@
+}
+
+function dropHead(entity: Entity, loc: Location) {
+  let drop: ItemStack | undefined;
+
+  // Get drop from entity
+  const key = entity.type.toString();
+  if (drops.has(key)) drop = drops.get(key);
+
+  // Get player head drop
+  if (entity instanceof Player) {
+    const player = entity as Player;
+    drop = new ItemStack(Material.PLAYER_HEAD, 1);
+    const meta = drop.itemMeta as SkullMeta;
+    meta.owningPlayer = player;
+    drop.itemMeta = meta;
+  }
+  if (drop) loc.world.dropItem(loc, drop);
+}
+
+async function playChopEffects(location: Location) {
+  new ParticleBuilder(Particle.BLOCK_DUST)
+    .location(location.clone().add(0, 0.5, 0))
+    .data(Material.REDSTONE_BLOCK.createBlockData())
+    .count(20)
+    .spawn();
+}
+
+// Drop blade on piston retract
+registerEvent(BlockPistonRetractEvent, (event) => {
+  const blocks = event.blocks;
+  for (const block of blocks) {
+    const entities = block.location.add(0, 1, 0).getNearbyEntities(1, 1, 1);
+    for (const entity of entities) {
+      if (entity.type === EntityType.ARMOR_STAND) {
+        const armorStand = entity as ArmorStand;
+        dropBlade(armorStand);
+      }
+    }
+  }
+});
+
+// Drop blade on block break
+registerEvent(BlockBreakEvent, (event) => {
+  const entities = event.block.location.add(0, 1, 0).getNearbyEntities(1, 1, 1);
+  for (const entity of entities) {
+    if (entity.type === EntityType.ARMOR_STAND) {
+      const armorStand = entity as ArmorStand;
+      dropBlade(armorStand);
+    }
+  }
+});
+
+// Place blade on block
+registerEvent(PlayerInteractEvent, (event) => {
+  if (!isRightClick(event.action)) return;
+  if (!event.clickedBlock) return;
+  if (event.hand !== EquipmentSlot.HAND) return;
+
+  const item = event.player.inventory.itemInMainHand;
+  if (!Blade.check(item)) return;
+
+  const player = event.player;
+  const block = event.clickedBlock.getRelative(event.blockFace);
+  const groundBlock = block.getRelative(BlockFace.DOWN);
+  const loc = groundBlock.location.toCenterLocation().add(0, 0.5, 0);
+
+  if (event.clickedBlock.type.isInteractable() && !player.isSneaking()) return;
+  if (block.type !== Material.AIR) return;
+  if (groundBlock.type === Material.AIR)
+    return player.sendActionBar('Et voi asettaa terää ilmaan.');
+
+  // Prevent placing inside entities
+  const entities = loc.getNearbyEntities(0.5, 0.5, 0.5);
+  if (!entities.isEmpty()) return;
+
+  loc.world.playSound(loc, PLACEMENT_SOUND, SoundCategory.BLOCKS, 0.6, 0.8);
+
+  // Prepare armor stand facing
+  switch (event.player.facing) {
+    case BlockFace.SOUTH:
+      loc.yaw = 180;
+      break;
+    case BlockFace.EAST:
+      loc.yaw = 90;
+      break;
+    case BlockFace.WEST:
+      loc.yaw = -90;
+      break;
+    default:
+      loc.yaw = 0;
+      break;
+  }
+
+  // Spawn armor stand out of sight
+  const armorStand = event.player.world.spawnEntity(
+    new Location(block.world, 0, -1000, 0),
+    EntityType.ARMOR_STAND,
+  ) as ArmorStand;
+
```

Comment:
Did we have common API for armorstand spawning? If not, you might want to teleport the armorstand X blocks away from the player, but still inside rendered chunks. Check placable-items for example (if the common API does not exist)

Reply:
At the moment, we only have it for item frames. Support for armor stands is planned but (probably) not very high priority.